### PR TITLE
use variables to toggle feature

### PR DESCRIPTION
**Pull request by nsa-yoda**
Creates flags to enable/disable features by setting them to true/false, allowing for easier configuration of features that a user might want. 

Also moves magic values to variables (like the wallpaper path, mouse speed, etc)

Screenshot 2024-08-26 at 2 24 52 PM

Screenshot 2024-08-26 at 2 26 03 PM

Screenshot 2024-08-26 at 2 26 11 PM

### CODE REVIEW

File: configure.sh [hunk 1]
```diff
@@ -1,37 +1,84 @@
 #!/bin/bash
 
+dark_mode=true
+dark_terminal=true
```
Comment (Tetrax-10) [inline configure.sh]: `dark_terminal` option is not necessary merge it with `dark_mode`.

File: configure.sh [hunk 2]
```diff
@@ -1,37 +1,84 @@
...
 # set auto sleep to never
-gsettings set org.gnome.desktop.session idle-delay 0
+if [ "$auto_sleep" = true ]; then
+  gsettings set org.gnome.desktop.session idle-delay 0
+fi
+
 # turn off automatic screen lock
-gsettings set org.gnome.desktop.screensaver lock-enabled false
+if [ "$dim_screen" = true ]; then
+  gsettings set org.gnome.desktop.screensaver lock-enabled false
```
Comment (Tetrax-10) [inline configure.sh:38]: why using `dim_screen` variable for this? shouldn't you be using something like `auto_screen_lock`?

File: configure.sh [hunk 3]
```diff
@@ -1,37 +1,84 @@
...
+dark_mode=true
+dark_terminal=true
+change_wallpaper=true
+wallpaper_path=/home/amnesia/Persistent/wallpapers/wallpaper.jpg
+auto_sleep=true
+dim_screen=true
+disable_camera=true
+disable_microphone=true
+disable_location=true
+disable_usb_auto_mount=false
```
Comment (Tetrax-10) [inline configure.sh]: the default value of `disable_usb_auto_mount` should be `true`.

**Review by Tetrax-10 — CHANGES_REQUESTED** [review]
Hey thanks for the PR, really nice one.

Please review the requested changes.

File: README.md [hunk 4]
```diff
@@ -4,6 +4,31 @@ A script to customize and secure Tails OS with privacy enhancements and UI tweak
...
+| Disable Dim Screen       | `disable_dim_screen`       | `true`        | Prevents automatic screen dimming.                  |
+| Disable Auto Screen Lock | `disable_auto_screen_lock` | `true`        | Prevents automatic locking of the screen.           |
+| Disable Camera           | `disable_camera`           | `true`        | Disables camera.                                    |
+| Disable Microphone       | `disable_microphone`       | `true`        | Disables microphone.                                |
+| Disable Location         | `disable_location`         | `true`        | Disables location services.                         |
+| Disable USB Auto Mount   | `disable_usb_auto_mount`   | `true`        | Disables automatic mounting of USB devices.         |
+| Disable File History     | `disable_file_history`     | `true`        | Disables the tracking of recent files.              |
+| Set Mouse Speed          | `set_mouse_speed`          | `false`       | Adjusts mouse speed.                                |
+| Disable Wi-Fi            | `disable_wifi`             | `false`       | Optionally disables Wi-Fi.                          |
```
Comment (Tetrax-10) [inline README.md]:
```suggestion
| Disable Wi-Fi            | `disable_wifi`             | `false`       | Disables Wi-Fi.                          |
```

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (Tetrax-10) [inline .gitignore]: `.gitignore` is not needed. Let the repo be simple with only required files.

```suggestion
```

**Review by Tetrax-10 — CHANGES_REQUESTED** [review]
I think everything is fine except these two minor changes.

**Comment by Tetrax-10** [comment]
I think I wasn't that clear. We don't need `.gitignore` file. Can you delete that file and commit?

**Comment by nsa-yoda** [comment]
> I think I wasn't that clear. We don't need `.gitignore` file. Can you delete that file and commit?

Ah not sure why the previous commit from your comment didn't apply. Updating.

**Review by Tetrax-10 — APPROVED** [review]

**Pull request merged**